src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java

```diff
@@ -81,6 +90,9 @@ protected void configurePolling(Graph graph, JsonNode config) throws Exception {
         if (config.path("fuzzyTripMatching").asBoolean(false)) {
             this.fuzzyTripMatcher = new GtfsRealtimeFuzzyTripMatcher(graph.index);
         }
+        this.connectionTimeout = config.path("connectionTimeoutMills").asInt();
+        this.connectionRequestTimeout = config.path("connectionRequestTimeoutMills").asInt();
+        this.socketTimeout = config.path("socketTimeoutMills").asInt();
```

Comment:
I'm wondering if we are backward-compatible with existing json configs, in case the parameters are not present?

Reply:
It should definitely be backward-compatible as it will fall back to default values if they are not specified.

Reply:
It took some poking around to determine which default values are applied. The documentation for asInt() states that a default value of zero is returned. These default zeros will then be passed into HttpUtils.getData(), which eventually passes them into getClient, where checks are performed to see that they're greater than zero. One of two things would need to be done: either we need Javadoc on every method that is called in this chain explaining the special significance of non-positive timeouts, or the defaults need to be applied at the point where the configuration is read.

Either way, the meaning of each of these timeouts needs to be explained in Javadoc (on the methods or fields) and in the user-facing documentation with example configuration files.

I do wonder whether it's necessary to expose such low level details to the average user. Could we expose a single timeout value rather than three separate timeouts?